src/webapp/components/dataset-table/DataSetTableConfig.tsx

```diff
@@ -196,10 +214,18 @@ export function getCommonActions<T extends ReferenceObject>(
             text: i18n.t("Logs"),
             icon: <ListIcon />,
             multiple: true,
-            isActive: props.isActive,
+            isActive: records => props.isActive(records) && user.isAdmin(),
             onClick(selectedIds) {
                 onAction({ ids: selectedIds, action: "logs" });
             },
         },
     ];
 }
+
+function canDataSetBeUpdated<T>(records: T[], user: User): boolean {
+    return records.every(dataSet => {
+        if (dataSet instanceof DataSetList) {
+            return dataSet.hasPermissionsToUpdate(user);
+        }
+    });
+}
```

Comment:
This is a bit off. Instead of simply T or T extends Ref, we could add an extra restriction, saying it has to support hasPermissionsToUpdate(user). And then add the method for the other classes that can be passed here. This way, from the moment you call the function you know what things are expected from the records passed. And then you can remove the instanceof check.